room/compiler/src/test/kotlin/androidx/room/processor/QueryMethodProcessorTest.kt

```diff
@@ -1106,19 +1124,13 @@ class QueryMethodProcessorTest(val enableVerification: Boolean) {
                     handler(null, parsedQuery, invocation)
                 }
             } else {
-                assertThat(adapter, nullValue())
+                assertThat(adapter, notNullValue())
             }
         }
-        if (enableVerification) {
-            return assertion
+        return if (enableVerification) {
+            assertion
         } else {
-            assertion.failsToCompile()
-                .withErrorContaining(
-                    cannotFindQueryResultAdapter(
-                        ClassName.get("foo.bar", "MyClass", "Pojo")
-                    )
-                )
-            return null
+            null
```

Comment:
before this `null` there needs to be a `assertion.compilesWithoutError()`, otherwise we are really not running compilation tests for when the verification is off